src/main/scala/io/iohk/ethereum/db/dataSource/DataSource.scala

```diff
@@ -0,0 +1,26 @@
+package io.iohk.ethereum.db.dataSource
+
+trait DataSource {
```

Comment:
What's the reason for not implementing an `insert` method separately from `update`? Just asking, please do not implement that :)